swift-tools-support-core/Sources/TSCBasic/FileSystem.swift

```diff
@@ -488,6 +504,9 @@ private class LocalFileSystem: FileSystem {
 //
 /// Concrete FileSystem implementation which simulates an empty disk.
 public class InMemoryFileSystem: FileSystem {
+    
+    /// Private internal representation of a file system node.
+    /// Not threadsafe.
```

Comment:
These differences are all from syncing the vendored TSC, right?  For historical reasons, it would be clearer to make the vendoring sync be a separate commit, I think.

Reply:
The vendoring sync is already separate in 27bd6a8f7bf815ac3f8dfbe3242f352473930a60, or is there anything that I missed?

Reply:
The subsequent SwiftPM update is in 68a729caa23d8b92585844def531ed98767bd5cc, and there's a small test fix separated in 6506c84ee38fd27a945df99376ca6f79a3f64dd0, but I can squash these last two if you'd like.

Reply:
Ah, right:  it was a mixture of familiar and unfamiliar diffs, but that was only because the new diffs are actually in TSC and so are being vendored along with the previous diffs.  I see what my mistake was there.  Please ignore the comment, I just misunderstood (I think I also expected the commit to be something like "syncing TSC" rather than "Improve FileSystemError by adding associated path", which made me think that it was something more than vendoring.

Maybe just squashing the new diffs here (unless you want them in the history) so that this can be merged without squashing would be good.  Basically, at the end it would be good to end up with one diff in the final history for vendoring TSC and another for the SwiftPM diffs.  Sorry for the extra trouble.

Reply:
No problem, thanks for the clarification. I've squashed the last two commits and changed the title of the first one to mention the vendored TSC copy sync. Let me know if anything else is needed here.